plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -243,20 +254,36 @@ private Map<String, ColumnHandle> makeColumnHandles(List<IndexMetadata.Field> fi
         for (BuiltinColumns builtinColumn : BuiltinColumns.values()) {
             result.put(builtinColumn.getName(), builtinColumn.getColumnHandle());
         }
-
         for (IndexMetadata.Field field : fields) {
             TypeAndDecoder converted = toTrino(field);
             result.put(field.name(), new OpenSearchColumnHandle(
-                    field.name(),
+                    ImmutableList.of(field.name()),
                     converted.type(),
                     field.type(),
                     converted.decoderDescriptor(),
-                    supportsPredicates(converted.type)));
+                    supportsPredicates(field.type(), converted.type())));
         }
 
         return result.buildOrThrow();
     }
 
+    private static boolean supportsPredicates(IndexMetadata.Type type, Type trinoType)
+    {
+        if (type instanceof DateTimeType) {
+            return true;
+        }
+        if (type instanceof ObjectType) {
```

Comment:
Why is this needed? Shouldn't it be handled by the `default` case in the switch below.

Reply:
I had to stop non primitive before it reaches the VarcharType but your suggestion https://github.com/trinodb/trino/pull/22646#discussion_r1716197598 helped.